src/focus-strategies/focus-by-registry.ts

```diff
@@ -0,0 +1,41 @@
+import { Direction, IArcHandler } from '../model';
+export class FocusByRegistry {
```

Comment:
I like the idea of popping strategies out to their own classes. I'd probably extend this slightly -- have it extend and abstract class and use Angular `multi`-providers to inject a list of strategies to the focus service, then loop through them.